### DISCUSSION TITLE
Schemas with same output type

### DISCUSSION THREAD

**Discussion by sdn90** [colinhacks/zod#803] [Q&A]
Hey all,

I have these 2 schemas:

```typescript
const BaseSchema = z.object({
  title: z.string(),
  author: z.string().optional(),
  seo: z.object({
    metaDescription: z.string()
  })
})

const PreviewBaseSchema = z.object({
  title: z.string().default("")
  author: z.string().optional()
  seo: z.object({
    metaDescription: z.string().default("")
  })
})
```

The output `PreviewSchema` will always be the same `BaseSchema`, but has different input requirements.

Is it possible to verify that the output of `PreviewSchema` is the same, including optional properties?

**Comment by JacobWeisenburger** [comment]
I don't think there is any built in function that lets you compare 2 schemas. But you might be able to use [lodash](https://lodash.com/) [_.isEqual](https://lodash.com/docs/#isEqual) or [_.isEqualWith](https://lodash.com/docs/#isEqualWith)
